File: php/categories.php
<?php

require_once 'entity.php';

class Category extends Entity
{
    public static $resource = '/categories/';
    public static $json_name = 'category';

    public function __construct()
    {
        $this->json_property('id','integer');
        $this->json_property('name','string');
        $this->json_property('show_dashboard','boolean');
        $this->json_property('show_public','boolean');
        $this->json_property('domain', 'Domain', $many = False,
                             $default = '1', $recursive = True);
    }
}

class Categories extends Resource
{
    public function __construct()
    {
        $this->entity_class = 'Category';
        $this->json_name = 'categories';
    }
}

$categories = new Categories();

File: python/categories.py
import sdk.python.entities
from sdk.python.domains import Domain
from sdk.python.entities import Property


class Category(sdk.python.entities.Entity):

    resource = '/categories/'
    json_name = 'category'

    id = Property(int)
    name = Property(str)
    show_dashboard = Property(bool)
    show_public = Property(bool)

    domain = Property(Domain, backref="categories")


class Categories(sdk.python.entities.Resource):

    entity_class = Category
    json_name = 'categories'


categories = Categories()
